template-processors/base/bin/processParameters.sh

```diff
@@ -21,45 +21,48 @@ echo "Processing Parameters"
 
 FOLDERS=""
 
-if [ -e ${CLONED_PARAMETER_GIT_DIR}/hiera.lst ] ; then
+if [ -e "${CLONED_PARAMETER_GIT_DIR}/hierarchy.lst" ] ; then
   echo "Generating hierarchy"
   # a hierarchy list was provided, so lets process it
-  while read -r DIR
-    do
+  while read -r DIR ; do
       # remove everything after # to allow comments and nuke all whitespaces
-      DIR="$(echo -e "${DIR%%#*}" | tr -d '[:space:]')"
+      DIR="$(echo "${DIR}" | sed -e 's/[[:space:]]*#.*//')"
 
       if [ -n "${DIR}" ]; then
         # add the folder to the end of the list
-        FOLDERS="${FOLDERS} ${CLONED_PARAMETER_GIT_DIR}/${DIR}"
+        DIR="$(cd ${CLONED_PARAMETER_GIT_DIR}/${DIR} ; pwd)"
+        FOLDERS="${FOLDERS} ${DIR}"
       fi
-    done
-fi <<< $(envsubst < ${CLONED_PARAMETER_GIT_DIR}/hiera.lst)
-
-# Add the current folder as the last one to process
-FOLDERS="${FOLDERS} ${CLONED_PARAMETER_GIT_DIR}"
-
-echo "Going to process parameters file in the following folder(s): ${FOLDERS}"
+    done <<< $(envsubst < ${CLONED_PARAMETER_GIT_DIR}/hierarchy.lst)
+else
+  # No hierarchy provided, so lets just use the current folder
+  FOLDERS="${CLONED_PARAMETER_GIT_DIR}"
+fi
 
-VALUES_FILE=${CLONED_PARAMETER_GIT_DIR}/eunomia_values_processed1.yaml
+# process the folders
+if [ -n "${FOLDERS}" ]; then
+  # ensure our base file exists with one document to make yq happy
+  VALUES_FILE="/tmp/eunomia_values_processed1.yaml"
+  echo "---" > ${VALUES_FILE}
 
-# ensure our base file exists with one document to make yq happy
-echo "---" > ${VALUES_FILE}
+  for DIR in ${FOLDERS}; do
+    echo "Processing files in ${DIR}"
 
-for DIR in ${FOLDERS}; do
-  echo "Processing files in ${DIR}"
-  
-  # get the list of yaml files to process
-  YAML_FILES="$(find ${DIR} -name \*.json -o -name \*.yaml -o -name \*.yml  -maxdepth 1)"
+    # get the list of yaml files to process
+    YAML_FILES="$(find ${DIR} -name \*.json -o -name \*.yaml -o -name \*.yml  -maxdepth 1)"
 
-  # merge the files
-  goyq merge --inplace ${VALUES_FILE} ${YAML_FILES}
-done
+    # merge the files
+    goyq merge -i -x ${VALUES_FILE} ${YAML_FILES}
+  done
+else
+  echo "ERROR - no folders found for processing"
+  exit 1
+fi
 
 # Replace variables from enviroment
 # This allows determining things like cluster names, regions, etc.
-if [ -e ${VALUES_FILE} ]; then
-  envsubst < ${VALUES_FILE} > $CLONED_PARAMETER_GIT_DIR/eunomia_values_processed.yaml
+if [ -e "${VALUES_FILE}" ]; then
+  envsubst < ${VALUES_FILE} > /tmp/eunomia_values_processed.yaml
```

Comment:
1. Please wrap env var in double quotes
2. Why the change from $CLONED_PARAMETER_GIT_DIR to /tmp/ ? the README seems to require processParameters.sh to put its output in $CLONED_PARAMETER_GIT_DIR ?

Reply:
1. done
2. moving it so we don't accidentally process it with the new behavior. fixed the readme.